### PR TITLE
remnote: refactor

### PR DESCRIPTION
**Pull request by chewblacka**
## Description of changes
- Updated version from 1.12.9 to 1.12.18
https://feedback.remnote.com/changelog/1-12-18

- Refactored code using stdenv to install an icon file and desktop file to the output path.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Please change your commit message(s) to match the [commit message conventions of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)!

File: pkgs/applications/misc/remnote/default.nix [hunk 1]
```diff
@@ -1,14 +1,45 @@
...
+    type = "Application";
+    name = "remnote";
+    desktopName = "RemNote";
+    comment = "Spaced Repetition";
+    icon = "remnote";
+    exec = "remnote %u";
+    categories = [ "Office" ];
+    mimeTypes = [ "x-scheme-handler/remnote" "x-scheme-handler/rn" ];
+  };
+in
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/remnote/default.nix:27]: Why aren't these variables inline in the `mkDerivation` call attributes?

File: pkgs/applications/misc/remnote/default.nix [hunk 2]
```diff
@@ -1,14 +1,45 @@
...
+  inherit pname version src;
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/bin
+    install ${appexec}/bin/remnote-${version} $out/bin/remnote
+    mkdir -p $out/share/applications
+    install -D "${desktopItem}/share/applications/"* -t $out/share/applications/
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/remnote/default.nix:38]: I believe that `desktopItem` is a valid attribute for `mkDerivation`, which would make that unnecessary. Could you check?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/251546/commits/dd0dbe3d8c4a4e98c7ae7d324c23d00d5f2820eb has a typo in the commit message. Care to fix that?

File: pkgs/applications/misc/remnote/default.nix (hunk 2 above)
Comment (chewblacka) [reply pkgs/applications/misc/remnote/default.nix:38]: > I believe that `desktopItem` is a valid attribute for `mkDerivation`, which would make that unnecessary. Could you check?

Hmm. I couldn't find any documentation to that effect, and my googling was fruitless. I tried adding it within `mkDerivation`, but couldn't get a desktop file to output. I basically copied the method used in the obsidian build file:

https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/misc/obsidian/default.nix

Comment (matthiasbeyer) [reply pkgs/applications/misc/remnote/default.nix:38]: Okay!

File: pkgs/applications/misc/remnote/default.nix [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
+  type = "Application";
+  name = "remnote";
+  desktopName = "RemNote";
+  comment = "Spaced Repetition";
+  icon = "remnote";
+  exec = "remnote %u";
+  categories = [ "Office" ];
+  mimeTypes = [ "x-scheme-handler/remnote" "x-scheme-handler/rn" ];
+};
+in
```
Comment (matthiasbeyer) [inline pkgs/applications/misc/remnote/default.nix:27]: Why are these variables not inlined in the `mkDerivation` call?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
https://github.com/NixOS/nixpkgs/pull/251546/commits/be5bedfde22c5f853aa897e0e2c23f8240149687 does not match the [commit message conventions of nixpkgs](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)!

**Pull request closed without merging**